### DISCUSSION TITLE
Label for select option of zero is not shown

### DISCUSSION THREAD

**Discussion by CherryDT** [payloadcms/payload#6084] [Feature Requests & Ideas]
### Describe the Bug

I have a field as follows:

```
            {
              name: 'spiciness',
              type: 'select',
              label: {
                de: 'Schärfegrad',
                en: 'Degree of Spiciness'
              },
              admin: {
                width: '25%'
              },
              required: true,
              options: [
                { value: 0, label: { de: 'Nicht scharf', en: 'Not spicy' } },
                { value: 1, label: { de: '🌶 Leicht scharf', en: '🌶 Mildly spicy' } },
                { value: 2, label: { de: '🌶🌶 Mittelscharf', en: '🌶🌶 Medium spicy' } },
                { value: 3, label: { de: '🌶🌶🌶 Sehr scharf', en: '🌶🌶🌶 Very spicy' } },
                { value: 4, label: { de: '🌶🌶🌶🌶 Extrem scharf', en: '🌶🌶🌶🌶 Extremely spicy' } }
              ]
            }
```

For some reason, it doesn't show the configured `label` of "Nicht scharf" for the first option with value zero:

![image](https://github.com/payloadcms/payload/assets/6010463/9c1ea045-2690-4141-9a71-174979edcc70)

### To Reproduce

Create a field as shown above and open the dropdown

### Payload Version

2.14.2

### Adapters and Plugins

db-mongodb, bundler-webpack

**Comment by jmikrut** [comment]
Hey there - well, we actually should be enforcing that select values are strings-only because of a variety of reasons.

One of which is that GraphQL `enum`s **_only_** allow string-based values.

In the future, if you do not want to use GraphQL, we could investigate ways to support number-based values in select / radio fields, but right now, this is not supported. I'll convert this into a Feature Request discussion where we can track interest in the support of number-based select values, knowing that they will not work as such in GraphQL / etc.

**Reply by CherryDT** [reply]
Ah I see. Thank you.

(I also stumbled upon a weird error when my value had a `:` in it, also due to GraphQL.)
